### ISSUE TITLE
list comprehensions don't see local variables in pdb in python3

### ISSUE THREAD

**Issue by 0f5d87d3-4357-4d19-af6f-dff73b145ded** [python/cpython#65360]
BPO | [21161](https://bugs.python.org/issue21161)
--- | :---
Nosy | @birkenfeld, @blueyed, @bitdancer, @florentx, @xdegaye
PRs | <li>python/cpython#15194</li>
Files | <li>[genexp.patch](https://bugs.python.org/file34791/genexp.patch "Uploaded as text/plain at 2014-04-12.09:28:15 by @xdegaye")</li><li>[default.patch](https://bugs.python.org/file35305/default.patch "Uploaded as text/plain at 2014-05-20.15:30:00 by @xdegaye")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2014-04-05.19:33:03.596>
created_at = <Date 2014-04-05.11:55:43.293>
labels = ['interpreter-core', '3.8', 'type-bug', '3.7', '3.9']
title = "list comprehensions don't see local variables in pdb in python3"
updated_at = <Date 2019-12-05.21:44:31.558>
user = 'https://bugs.python.org/kay'
```

bugs.python.org fields:
```python
activity = <Date 2019-12-05.21:44:31.558>
actor = 'inducer'
assignee = 'none'
closed = True
closed_date = <Date 2014-04-05.19:33:03.596>
closer = 'georg.brandl'
components = ['Interpreter Core']
creation = <Date 2014-04-05.11:55:43.293>
creator = 'kay'
dependencies = []
files = ['34791', '35305']
hgrepos = []
issue_num = 21161
keywords = ['patch']
message_count = 10.0
messages = ['215595', '215625', '215876', '215963', '215966', '215967', '216033', '216171', '218847', '348829']
nosy_count = 8.0
nosy_names = ['georg.brandl', 'blueyed', 'inducer', 'r.david.murray', 'flox', 'xdegaye', 'Jurko.Gospodneti\xc4\x87', 'kay']
pr_nums = ['15194']
priority = 'normal'
resolution = 'wont fix'
stage = None
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue21161'
versions = ['Python 3.1', 'Python 3.2', 'Python 3.3', 'Python 3.4', 'Python 3.5', 'Python 3.6', 'Python 3.7', 'Python 3.8', 'Python 3.9']
```

</p></details>

**Comment by hstojic** [comment]
will this be fixed at any point? its really a nuisance when debugging

**Comment by gvanrossum** [comment]
This came up in https://discuss.python.org/t/inlining-comprehensions-what-are-the-compatibility-constraints/22527/25

Reopening.

**Comment by carljm** [comment]
#101441 (PEP 709) will fix this for list, dict, and set comprehensions, but not for generator expressions.

**Comment by gaogaotiantian** [comment]
Fixed by #111094
